docs/wiki/EL/el-architecture.md

```diff
@@ -112,15 +112,32 @@ The beacon chain  invokes the new payload function and transfers the execution
 
 ##### Geth
 
-TODO: STF code links and walk through in Geth
+###### Transaction Execution in Geth
+Geth, like other Ethereum execution clients, processes transactions by verifying signatures, checking nonces, deducting gas fees, and updating the state. Transactions first enter the mempool, where they wait to be included in a block. Once picked up, Geth executes them, modifying account balances, contract storage, and other state data.
 
-Check week 2 talk by lightclient for an overview.
+🔗[ Transaction Execution Code](https://github.com/ethereum/execution-specs/blob/0f9e4345b60d36c23fffaa69f70cf9cdb975f4ba/src/ethereum/shanghai/fork.py#L542)
+
+###### Block Processing & State Updates
+Every new block contains multiple transactions that Geth processes in order. Once all transactions are executed, the final state is committed, and a state root hash is stored to ensure consistency. This process follows a defined set of rules to maintain network integrity.
+
+🔗 [State Transition Code](https://github.com/ethereum/execution-specs/blob/0f9e4345b60d36c23fffaa69f70cf9cdb975f4ba/src/ethereum/shanghai/fork.py#L145)
+
+###### Networking & Peer-to-Peer Communication
+Ethereum nodes communicate using DevP2P, a protocol that allows execution clients to exchange transactions and blocks. When a new transaction is sent, it propagates across the network through peer-to-peer connections, ensuring all nodes remain in sync. Each recipient verifies the transaction before forwarding it, preventing spam and invalid state transitions.
+
+🔗[DevP2P Specification](https://github.com/ethereum/devp2p/blob/master/discv4.md)
```

Comment:
The link here is to discv4, which handles peer discovery over UDP. It helps nodes find and maintain peer connections. It seems like this section is talking more about how nodes gossip tsx though, which happens via the Ethereum Wire Protocol after peers are connected via discv4. These tsx propogate via TCP, using RLPx. Link for ethereum wire protocol: https://github.com/ethereum/devp2p/blob/master/caps/eth.md